test/example/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 
 ### Example application
 
-add_executable(llvm-dialects-example
+add_executable(llvm-dialects-example EXCLUDE_FROM_ALL
```

Comment:
You can also put the `EXCLUDE_FROM_ALL` onto the [`add_subdirectory`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) for the example folder, then you don’t need to add it to every target created here.